src/common/utxobased/keymanager/bip69.ts

```diff
@@ -0,0 +1,54 @@
+// Previous transaction hashes (in reversed byte-order) are to be sorted in ascending order, lexicographically.
+// In the event of two matching transaction hashes, the respective previous output indices will be compared by their integer value, in ascending order.
+
+import { Input, Output } from './utxopicker'
+
+// If the previous output indices match, the inputs are considered equal.
+const inputComparator = (
+  aHash: string | Buffer,
+  aVout: number,
+  bHash: string | Buffer,
+  bVout: number
+): number => {
+  if (typeof aVout !== 'number' || typeof bVout !== 'number')
+    throw new TypeError('Expected vouts of type Number')
+  if (typeof aHash === 'string' && typeof bHash === 'string') {
+    const compare = aHash.localeCompare(bHash)
+    return compare !== 0 ? compare : aVout - bVout
+  }
+
+  if (!Buffer.isBuffer(aHash) || !Buffer.isBuffer(bHash))
+    throw new TypeError('Expected hashes of type Buffer')
+
+  const aHashR = ([].reverse.call(Buffer.from(aHash)) as unknown) as Buffer
+  const bHashR = ([].reverse.call(Buffer.from(aHash)) as unknown) as Buffer
+  const compare = aHashR.compare(bHashR)
+  return compare !== 0 ? compare : aVout - bVout
+}
+
+// Transaction output amounts (as 64-bit unsigned integers) are to be sorted in ascending order.
+// In the event of two matching output amounts, the respective output scriptPubKeys (as a byte-array) will be compared lexicographically, in ascending order.
+// If the scriptPubKeys match, the outputs are considered equal.
+const outputComparator = (
+  aScript: Buffer,
+  aValue: number,
+  bScript: Buffer,
+  bValue: number
+): number => {
+  if (typeof aValue !== 'number' || typeof bValue !== 'number')
+    throw new TypeError('Expected values of type Number')
+
+  return aValue === bValue ? aScript.compare(bScript) : aValue - bValue
+}
+
+export const sortInputs = (inputs: Input[]): Input[] => {
+  return inputs.concat().sort(function (a, b) {
```

Comment:
What is the `concat` call for?

Reply:
Creates a new array with the same elements. The same as [... inputs].
That's why I rewrote it :)